core/src/test/java/com/datastax/oss/driver/internal/core/config/typesafe/DefaultDriverConfigLoaderTest.java

```diff
@@ -174,4 +178,48 @@ public void should_not_notify_from_manual_reload_if_config_has_not_changed() {
     verify(eventBus, never()).fire(ConfigChangeEvent.INSTANCE);
     assertThatStage(reloaded).isSuccess(changed -> assertThat(changed).isFalse());
   }
+
+  @Test
+  public void should_load_from_other_classpath_resource() {
+    DriverConfigLoader loader = DriverConfigLoader.fromClasspath("config/customApplication");
+    DriverConfig config = loader.getInitialConfig();
+    // From customApplication.conf:
+    assertThat(config.getDefaultProfile().getDuration(DefaultDriverOption.REQUEST_TIMEOUT))
+        .isEqualTo(Duration.ofSeconds(5));
+    // From customApplication.json:
+    assertThat(config.getDefaultProfile().getInt(DefaultDriverOption.REQUEST_PAGE_SIZE))
+        .isEqualTo(2000);
+    // From customApplication.properties:
+    assertThat(config.getDefaultProfile().getString(DefaultDriverOption.REQUEST_CONSISTENCY))
+        .isEqualTo(DefaultConsistencyLevel.ONE.name());
+    // From reference.conf:
+    assertThat(config.getDefaultProfile().getString(DefaultDriverOption.REQUEST_SERIAL_CONSISTENCY))
+        .isEqualTo(DefaultConsistencyLevel.SERIAL.name());
+  }
+
+  @Test
+  public void should_load_from_file() {
+    File file = new File("src/test/resources/config/customApplication.conf");
+    DriverConfigLoader loader = DriverConfigLoader.fromFile(file);
+    DriverConfig config = loader.getInitialConfig();
+    // From customApplication.json:
+    assertThat(config.getDefaultProfile().getDuration(DefaultDriverOption.REQUEST_TIMEOUT))
+        .isEqualTo(Duration.ofSeconds(5));
+    // From reference.conf:
+    assertThat(config.getDefaultProfile().getString(DefaultDriverOption.REQUEST_SERIAL_CONSISTENCY))
+        .isEqualTo(DefaultConsistencyLevel.SERIAL.name());
+  }
+
+  @Test
+  public void should_load_from_url() throws Exception {
+    URL url = new File("src/test/resources/config/customApplication.conf").toURI().toURL();
```

Comment:
From what I saw in the TypeSafe code this is not really exercising URLs because file URLs are special-cased, but not sure if we really want to got as far as creating other types of URLs for tests. Maybe we could create a special `URLStreamHandler` for a `foo` scheme? Or use WireMock?

Reply:
> From what I saw in the TypeSafe code this is not really exercising URLs

Good catch.
Yeah I'm not sure if it's worth starting a web container from the test just for this. It's not even our code. I think I'll just do a quick manual test to validate that it works, and then we can trust [Config's own tests](https://github.com/lightbend/config/blob/master/config/src/test/scala/com/typesafe/config/impl/HttpTest.scala).